File: ClientApp/src/app/api/models/ore-location.ts
/* tslint:disable */
/* eslint-disable */
import { Location } from '../models/location';
import { LocationType } from '../models/location-type';
import { OreLocationElement } from '../models/ore-location-element';
export interface OreLocation {
  id?: number;
  locationType?: LocationType;
  name?: null | string;
  ores?: null | Array<OreLocationElement>;
  rootLocation?: Location;
}

File: Models/OreLocation.cs
#nullable disable

using System.Text.Json.Serialization;

namespace WebPage.Models
{
    public class OreLocation
    {
        public int Id { get; set; }
        public Location RootLocation { get; set; } = new Location();
        [JsonPropertyName("name")]
        public string name { get; set; }
        [JsonPropertyName("locationType")]
        public LocationType LocationType { get; set; }
        [JsonPropertyName("ores")]
        public List<OreLocationElement> Ores { get; set; } = new List<OreLocationElement>();
    }

}
